src/Service/PdfCreator.php

```diff
@@ -4,38 +4,32 @@
 
 namespace App\Service;
 
-use Spipu\Html2Pdf\Html2Pdf;
+use App\ValueObject\PdfFormat;
+use App\ValueObject\PdfOrientation;
+use Dompdf\Dompdf;
+use Dompdf\Options;
 
 final class PdfCreator implements PdfCreatorInterface
 {
     private $pdf;
     private $orientation;
     private $format;
-    private $lang;
-    private $unicode;
-    private $encoding;
-    private $margin;
+    private $options;
 
-    public function __construct(string $orientation, string $format, string $lang, bool $unicode, string $encoding, array $margin)
+    public function __construct(PdfOrientation $orientation, PdfFormat $format, array $options)
     {
         $this->orientation = $orientation;
         $this->format = $format;
-        $this->lang = $lang;
-        $this->unicode = $unicode;
-        $this->encoding = $encoding;
-        $this->margin = $margin;
+        $this->options = $options;
     }
 
-    public function create(): void
+    public function generatePdf(string $template): string
     {
-        $this->pdf = new Html2Pdf($this->orientation, $this->format, $this->lang, $this->unicode, $this->encoding, $this->margin);
-        $this->pdf->setTestIsImage(false);
-    }
-
-    public function generatePdf(string $template, string $name): string
-    {
-        $this->pdf->writeHTML($template);
+        $this->pdf = new Dompdf((new Options())->set($this->options));
```

Comment:
If `$this->pdf` is only used locally to this method, why having it as an attribute? Also, I would consider creating the instance and configuring it in the constructor I guess.
I'd probably have Options() (with default values) injected, and an optional empty array of options passed for generation afterward.